apps/examples/reboot_reason_test/reboot_reason_main.c

```diff
@@ -0,0 +1,225 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#ifdef CONFIG_MM_ASSERT_ON_FAIL
+#include <stdlib.h>
+#endif
+#ifdef CONFIG_WATCHDOG
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <tinyara/watchdog.h>
+#endif
+#ifdef CONFIG_LIB_BOARDCTL
+#include <sys/boardctl.h>
+#endif
+#include <sys/prctl.h>
+#include <tinyara/reboot_reason.h>
+
+#define REBOOT_REASON_TEST (REBOOT_UNKNOWN - 1)
+
+static void display_reboot_reason_option(void)
+{
+	printf("\nSelect Reboot Reason Test Option.\n");
+	printf("\t-Press H or h : HW Reset Test\n");
+#ifdef CONFIG_MM_ASSERT_ON_FAIL
+	printf("\t-Press M or m : Memory Allocation Fail Test\n");
+#endif
+#ifdef CONFIG_WATCHDOG
+	printf("\t-Press W or w : Watchdog Reset Test\n");
+#endif
+	printf("\t-Press V or v : Random Value Write-Read Test\n");
+	printf("\t-Press C or c : Clear reboot reason\n");
+	printf("\t-Press R or r : Check previous reboot reason\n");
+	printf("\t-Press X or x : Terminate Tests.\n");
+}
+#ifdef CONFIG_WATCHDOG
+static void watchdog_test(void)
+{
+	int fd;
+	int ret;
+	/* Test for Watchdog Reset */
+	printf("After watchdog reset, Expected Reboot reason is %d\n", REBOOT_SYSTEM_WATCHDOG);
+
+	fd = open(CONFIG_WATCHDOG_DEVPATH, O_RDWR);
+	if (fd < 0) {
+		printf("ERROR: Fail to open %s\n", CONFIG_WATCHDOG_DEVPATH);
+		return;
+	}
+
+	ret = ioctl(fd, WDIOC_SETTIMEOUT, 1000);
+	if (ret != OK) {
+		printf("ERROR: Fail to set watchdog timeout.\n");
+		close(fd);
+		return;
+	}
+
+	ret = ioctl(fd, WDIOC_START, 0);
+	if (ret != OK) {
+		printf("ERROR: Fail to start watchdog.\n");
+		close(fd);
+		return;
+	}
+}
```

Comment:
Don't you need to wait until watchdog time out?

Reply:
Changed to wait sem. Thanks.